plugins/ecs-bridge/engine/engine.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package engine
+
+import (
+	"net"
+	"syscall"
+
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/cniipamwrapper"
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/cniipwrapper"
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/cninswrapper"
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/netlinkwrapper"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+)
+
+// Engine represents the execution engine for the ECS Bridge plugin.
+// It defines all the operations performed during the execution of the
+// plugin
+type Engine interface {
```

Comment:
The workflow that these methods represent are fairly obvious and simple to understand IMO:
1. Create bridge
2. Create veth pair
3. Assign the other end to the container
4. Assign one end of veth pair to the bridge

Between steps 3 & 4, configure the container's veth pair with ip addresses. I would not like to abstract `veth pairs` with the term `port`, because a port on linux can also refer to the port number used for socket communication.